lib/src/unify_package_rename_suggestors/import_renamer.dart

```diff
@@ -0,0 +1,105 @@
+      // Most likely a part and not a library.
+      return;
+    }
+
+    // Parts that have not been generated can show up as `exists = false` but also `isPart = false`,
+    // so using the unitResults is a little trickier than using the libraryElement to get it.
+    final mainLibraryUnitResult = libraryResult.units.singleWhere(
+        (unitResult) =>
+            unitResult.unit.declaredElement ==
+            libraryResult.element.definingCompilationUnit);
+
+    // Look for imports with old package name.
+    final importsToUpdate = mainLibraryUnitResult.unit.directives
+        .whereType<ImportDirective>()
+        .where((import) =>
+            import.uri.stringValue?.startsWith('package:$oldPackageName/') ??
+            false);
+
+    final newImportsInfo = <UnifyImportInfo>[];
+    for (final import in importsToUpdate) {
+      final importUri = import.uri.stringValue;
+      final namespace = import.prefix?.name;
+      var newImportUri = importUri?.replaceFirst(
+          'package:$oldPackageName/', 'package:$newPackageName/');
+      var newNamespace =
+          namespace == oldPackageNamespace ? newPackageNamespace : namespace;
+
+      // Check for special cases where the unify_ui import path does not match the previous RMUI path.
+      final specialCaseRmuiImport =
+          rmuiImportsToUpdate.where((i) => importUri == i.rmuiUri);
+      if (specialCaseRmuiImport.isNotEmpty) {
+        newImportUri = specialCaseRmuiImport.single.uri;
+
+        final specialCaseNamespace = specialCaseRmuiImport.single.namespace;
+        if (namespace != null &&
+            specialCaseNamespace != null &&
+            (specialCaseRmuiImport.single.possibleMuiNamespaces
+                    ?.contains(namespace) ??
+                false)) {
+          newNamespace = specialCaseNamespace;
+        }
+      }
+
+      if (newImportUri != null) {
+        // Collect info on new imports to add.
+        newImportsInfo
+            .add(UnifyImportInfo(newImportUri, namespace: newNamespace));
+      }
+
+      final prevTokenEnd = import.beginToken.previous?.end;
+      // Try to take the newline before the import, but watch out
+      // for prevToken's offset/end being -1 if it's this import has the
+      // first token in the file.
+      final startOffset = prevTokenEnd != null && prevTokenEnd != -1
+          ? prevTokenEnd
+          : import.offset;
+      yield Patch('', startOffset, import.end);
+    }
+
+    // Sort imports before adding them.
+    newImportsInfo.sort((a, b) => a.uri.compareTo(b.uri));
+
+    // Add imports in their alphabetical positions.
+    for (final importInfo in newImportsInfo) {
+      final insertInfo = insertionLocationForPackageImport(importInfo.uri,
+          mainLibraryUnitResult.unit, mainLibraryUnitResult.lineInfo);
+      yield Patch(
+          insertInfo.leadingNewlines +
+              "import '${importInfo.uri}'${importInfo.namespace != null ? ' as ${importInfo.namespace}' : ''};" +
```

Comment:
just throwing this out there that this doesnt seem to retain `show` statements 

example:
`import 'package:react_material_ui/styles/theme_provider.dart' as mui_theme show UnifyThemeProvider;`

Reply:
Ooh good catch! Yeah I think it'd be worth updating so this retains `show` and `hide`

Reply:
Fixed!